crates/matrix-sdk-base/src/client.rs

```diff
@@ -803,7 +808,8 @@ impl BaseClient {
         Ok(response)
     }
 
-    async fn apply_changes(&self, changes: &StateChanges) {
+    #[tracing::instrument]
```

Comment:
Can all these tracing additions be moved into a separate PR? This one is already huge.

Reply:
I needed this for debugging in between, but arguably you are right, all that should be out of here as it is unrelated.